src/tagmanager/tm_parser.c

```diff
@@ -1531,38 +1531,32 @@ gchar *tm_parser_format_function(TMParserType lang, const gchar *fname, const gc
 
 	if (retval)
 	{
+		const gchar *sep = NULL;
+
 		switch (lang)
 		{
-			case TM_PARSER_GDSCRIPT:
-			case TM_PARSER_GO:
+			/* retval after function */
 			case TM_PARSER_PASCAL:
+				sep = ": ";
+				break;
+			case TM_PARSER_GDSCRIPT:
 			case TM_PARSER_PYTHON:
-			{
-				/* retval after function */
-				const gchar *sep;
-				switch (lang)
-				{
-					case TM_PARSER_PASCAL:
-						sep = ": ";
-						break;
-					case TM_PARSER_GDSCRIPT:
-					case TM_PARSER_PYTHON:
-						sep = " -> ";
-						break;
-					default:
-						sep = " ";
-						break;
-				}
-				g_string_append(str, sep);
-				g_string_append(str, retval);
+				sep = " -> ";
+				break;
+			case TM_PARSER_GO:
+				sep = " ";
 				break;
-			}
 			default:
 				/* retval before function */
 				g_string_prepend_c(str, ' ');
 				g_string_prepend(str, retval);
 				break;
 		}
+		if (sep)
+		{
+			g_string_append(str, sep);
+			g_string_append(str, retval);
+		}
```

Comment:
I'd rather have an if/else here if the value of `sep` is the criteria, like that I find the code weird as well.
```suggestion
				break;
		}

		if (sep)
		{
			/* retval after function */
			g_string_append(str, sep);
			g_string_append(str, retval);
		}
		else
		{
			/* retval before function */
			g_string_prepend_c(str, ' ');
			g_string_prepend(str, retval);
		}
```